app/controllers/balances_controller.rb

```diff
@@ -1,7 +1,16 @@
 class BalancesController < ApplicationController
-  def show
+  def def index
+    if params[:date_from].blank? and params[:data_to].blank?
+      @transactions = BankTransaction.all
+    elsif !params[:data_from].blank? and !params[:data_to].blank?
+      @transactions = BankTransaction.get_transactions_for_time_period(params[:data_from], params[:data_to])
+    else
+      render :nothing => true, :status => :bad_request 
+    end
```

Comment:
Ich würde dem nutzer das formular nochmals an den Kopf werfen, anstelle dass er nichts sieht.

`:nothing` ist wirklich nichts, also alles weiß

```suggestion
  def show
    return render(:new) unless params[:start_date] && params[:end_date]
    
    @transactions = BankTransaction.for_time_period(params[:start_date], params[:end_date])
    @balance = Balance.new(@transactions, nil) # TODO add bank account
  end
```